Essentials/src/net/ess3/api/events/UserBalanceUpdateEvent.java

```diff
@@ -48,4 +55,16 @@ public void setNewBalance(BigDecimal newBalance) {
     public BigDecimal getOldBalance() {
         return originalBalance;
     }
+
+    public Cause getCause() {
+        return cause;
+    }
+
+    public enum Cause {
+        COMMAND_ECO,
+        COMMAND_PAY,
+        COMMAND_SELL,
+        SPECIAL, // Reserved for API usage
+        UNKNOWN
```

Comment:
Perhaps we should include a cause such as `API` represents updates triggered by legacy API (`com.earth2me.essentials.api.Economy`) and Vault (which uses the `cee.api.Economy` class anyway)?